packages/PlanLimitsUI/src/lib/useDetailedWaterUseData.ts

```diff
@@ -0,0 +1,122 @@
+import { useWaterUseQuery } from '../api';
+import {
+  format,
+  addYears,
+  addDays,
+  getWeek,
+  parse,
+  lastDayOfWeek,
+} from 'date-fns';
+import { groupBy, flatten } from 'lodash';
+import { GWUsagePresentationGroups } from '../lib/councilData';
+
+interface ParsedUsage extends Usage {
+  parsedDate: Date;
+  endOfWeek: Date;
+}
+
+export default function useDetailedWaterUseData(councilId: number) {
+  const to = addDays(new Date(), -1);
+  const from = addYears(to, -1);
+  const formattedFrom = format(from, 'yyyy-MM-dd');
+  const formattedTo = format(to, 'yyyy-MM-dd');
+  const waterUseQueryResult = useWaterUseQuery(
+    councilId,
+    formattedFrom,
+    formattedTo,
+  );
+
+  const data = {
+    from,
+    to,
+    formattedFrom,
+    formattedTo,
+    usage: waterUseQueryResult.data
+      ? transformWaterUseData(waterUseQueryResult.data)
+      : undefined,
+  };
+
+  return {
+    ...waterUseQueryResult,
+    data,
+  };
+}
+
+export type DetailedWaterUseQuery = ReturnType<typeof useDetailedWaterUseData>;
+
+function transformWaterUseData(usage: Usage[]) {
+  const parsedUsage = parseUsage(usage);
+  const areaUsage = groupBy<ParsedUsage>(parsedUsage, 'areaId');
+
+  const groupedAreaUsage = GWUsagePresentationGroups.map((group) => {
```

Comment:
Guessing you might have thought of this already, but hard coded to this GW model even though in the function that calls this, we pass in the council id. 
Just wondering if we should either hard code them both in the same place ... or pass them both in

Reply:
Good point, have integrated it properly with CouncilData [here](https://github.com/Greater-Wellington-Regional-Council/Environmental-Outcomes-Platform/pull/148/commits/660db4b97ebc1f552f5a4bfad2261fb2f755e493).